docs/reference/sql/data-definition.md

```diff
@@ -0,0 +1,228 @@
+---
+layout: page
+title: Data definition
+tagline: Use DDL to structure data 
+description: How to use DDL to structure data in ksqlDB
+keywords: ksqldb, sql, ddl
+---
+
+This section covers how you create the structures that store your events.
+ksqlDB abstracts events as rows with columns and stores them in streams
+and tables.
+
+## Rows and columns
+
+Streams and tables help you model collections of events that accrete over time.
+Both are represented as a series of rows and columns with a schema, much like a
+relational database table. Rows represent individual events. Columns represent
+the attributes of those events.
+
+Each column has a data type. The data type limits the span of permissible values
+that you can assign. For example, if a column is declared as type `INT`, it can't
+be assigned the value of string `'foo'`.
+
+In contrast to relational database tables, the columns of a row in ksqlDB are
+divided into _key_ and _value_ columns. The key columns control which partition
+a row resides in. The value columns, by convention, store the main data of
+interest. Controlling the key columns is useful for manipulating the underlying
+data locality, and enables you to integrate with the wider {{ site.ak }}
+ecosystem, which uses the same key/value data model. By default, a column is a
+value column. Marking a column as a `(PRIMARY) KEY` makes it a key column.
+
+Internally, each row is backed by a [Kafka record](../../../overview/apache-kafka-primer/#records).
+In {{ site.ak }}, the key and value parts of a record are
+[serialized](../../../overview/apache-kafka-primer/#serializers) independently.
+ksqlDB enables you to exercise this same flexibility and builds on the semantics
+of {{ site.ak }} records, rather than hiding them.
+
+There is no theoretical limit on the number of columns in a stream or table.
+In practice, the limit is determined by the maximum message size that {{ site.ak }}
+can store and the resources dedicated to ksqlDB.
+
+## Streams
+
+A stream is a partitioned, immutable, append-only collection that represents a
+series of historical facts. For example, the rows of a stream could model a
+sequence of financial transactions, like "Alice sent $100 to Bob", followed by
+"Charlie sent $50 to Bob".
+
+Once a row is inserted into a stream, it can never change. New rows can be
+appended at the end of the stream, but existing rows can never be updated or
+deleted.
+
+Each row is stored in a particular partition. Every row, implicitly or explicitly,
+has a key that represents its identity. All rows with the same key reside in the
+same partition.
```

Comment:
> Every row, implicitly or explicitly, has a key that represents its identity.

This isn't true - ksqlDB supports keyless streams.  Keyless streams ignore the key of the source and produce to the output with no key set (meaning rows are randomly assigned a partition).